src/map/core/MapView.jsx

```diff
@@ -118,9 +118,15 @@ const MapView = ({ children }) => {
     };
   }, [containerEl]);
 
+  
   return (
     <div style={{ width: '100%', height: '100%' }} ref={containerEl}>
-      {mapReady && children}
+      {React.Children.map(children, child => {
+        if (React.isValidElement(child) && child.type.handlesMapReady) {
+          return React.cloneElement(child, { mapReady });
```

Comment:
Would this cause element to re-render all the time?

Reply:
The only change I can see to the rendering in the profiler is the desired behaviour. When mapReady goes to false the "handlesMapReady" children render themselves rather than disappearing from the tree, then when mapReady goes to true all the components re-render - some for the first time and some due to mapReady change.